src/renderer/src/routes/Onboarding/CreateProjectScreen.tsx

```diff
@@ -107,52 +108,64 @@ const HorizontalLine = styled('div')({
 function CreateProjectScreenComponent() {
 	const navigate = useNavigate()
 	const { formatMessage } = useIntl()
+
 	const [projectName, setProjectName] = useState('')
-	const [error, setError] = useState(false)
+	const [hasNameError, setHasNameError] = useState(false)
 	const [errorMessage, setErrorMessage] = useState('')
-	const setProjectNameMutation = useCreateProject()
+	const [configPath, setConfigPath] = useState<string | null>(null)
+
+	const createProjectMutation = useCreateProject()
+	const selectConfigFile = useSelectProjectConfigFile()
 	const { setActiveProjectId } = useActiveProjectIdStoreActions()
 
-	const [configFileName, setConfigFileName] = useState<string | null>(null)
+	function handleImportConfig() {
+		selectConfigFile.mutate(undefined, {
+			onSuccess: (filePath) => {
+				if (filePath) {
+					setConfigPath(filePath)
+				}
+			},
+			onError: (err) => {
+				console.error('Error selecting file', err)
+			},
+		})
+	}
 
 	const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
 		const value = event.target.value
+		setProjectName(value)
+
 		const localError: boolean = checkForError(
 			value.trim(),
 			PROJECT_NAME_MAX_LENGTH_GRAPHEMES,
 		)
-		setProjectName(value)
-		if (localError !== error) {
-			setError(localError)
-		}
-		setError(localError)
+		setHasNameError(localError)
 	}
 
 	const graphemeCount = countGraphemes(projectName.trim())
 
-	const handleAddName = () => {
+	const handleCreateProject = () => {
 		if (checkForError(projectName, PROJECT_NAME_MAX_LENGTH_GRAPHEMES)) {
-			setError(true)
+			setHasNameError(true)
 			return
 		}
-		setProjectNameMutation.mutate(projectName, {
-			onSuccess: (projectId) => {
-				setActiveProjectId(projectId)
-				navigate({ to: '/tab1' })
-			},
-			onError: (error) => {
-				console.error('Error setting project name:', error)
-				setErrorMessage(formatMessage(m.errorSavingProjectName))
-			},
-		})
-	}
 
-	function importConfigFile() {
-		// Placeholder for file import logic
-		setConfigFileName('myProjectConfig.comapeocat')
+		createProjectMutation.mutate(
+			{ name: projectName.trim(), configPath: configPath ?? undefined },
+			{
+				onSuccess: (projectId) => {
+					setActiveProjectId(projectId)
+					navigate({ to: '/tab1' })
+				},
+				onError: (error) => {
+					console.error('Error saving project:', error)
+					setErrorMessage(formatMessage(m.errorSavingProjectName))
```

Comment:
saving a project can also fail if the config file is invalid e.g. you selected a file with the correct extension but it's not actually the correct format.

Quick suggestion is to update the surfaced error text as something that's more generic and not specifically about the name.